File: 02_BFS/Gyeongwook/reverseInteger.py
class Solution:
    # @return an integer
    def reverse(self, x):
        result = 0
        source = x
        sign = 1
        
        if x < 0 :
            sign = -1
            source = ~x + 1
        
        while source is not 0 :
            result = result * 10
            result = result + (source % 10)
            source = source / 10
            
        result = sign * result
        
        if result > 2147483647 :
            return 0
        elif result < -2147483648 :
            return 0
            
        return result

File: 02_BFS/Eunbeom/Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        
        long long result = 0;
        int flag = 1;
        
        if(x<0){
            flag = -1;
            x = -x;
        }
    
        for( ;x>0; x/=10)
            result = result * 10 + x % 10;
        result *= flag;
        
        if(result > INT_MAX || result < INT_MIN) return 0;
        return result;
    }
};